Makefile

```diff
@@ -89,31 +87,19 @@ image:
 push:
 	docker push ${IMG}
 
-# Download controller-gen locally if necessary
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+
+.PHONY: controller-gen
 CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
-controller-gen:
-	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
+controller-gen: # Download controller-gen locally if necessary
+	GOBIN=$(PROJECT_DIR)/bin go install -mod=readonly sigs.k8s.io/controller-tools/cmd/controller-gen@latest
```

Comment:
Suggest to pin `controller-gen` and `setup-envtest` versions

Reply:
or better extract this versions into variables and move it to start of makefile

Reply:
done!